luminoth/models/ssd/ssd.py

```diff
@@ -0,0 +1,338 @@
+
+        # This is the dict we'll return after filling it with SSD's results
+        prediction_dict = {}
+
+        # Generate targets for training
+        if is_training and gt_boxes is not None:
+            gt_boxes = tf.cast(gt_boxes, tf.float32)
+
+            # Generate targets
+            target_creator = SSDTarget(
+                self._num_classes, anchors.shape[0], self._config.model.target,
+                self._config.model.variances
+            )
+            class_targets, bbox_offsets_targets = target_creator(
+                class_probabilities, anchors, gt_boxes,
+                tf.cast(tf.shape(image), tf.float32)
+            )
+
+            # Filter the predictions and targets that we will ignore
+            # during training due to hard negative mining.
+            # We use class_targets to know which ones to ignore (they
+            # are marked as -1 if they are to be ignored)
+            with tf.name_scope('prepare_batch'):
+                predictions_filter = tf.greater_equal(class_targets, 0)
+
+                anchors = tf.boolean_mask(
+                    anchors, predictions_filter)
+                bbox_offsets_targets = tf.boolean_mask(
+                    bbox_offsets_targets, predictions_filter)
+                class_targets = tf.boolean_mask(
+                    class_targets, predictions_filter)
+                class_scores = tf.boolean_mask(
+                    class_scores, predictions_filter)
+                class_probabilities = tf.boolean_mask(
+                    class_probabilities, predictions_filter)
+                bbox_offsets = tf.boolean_mask(
+                    bbox_offsets, predictions_filter)
+
+            # Add target tensors to prediction dict
+            prediction_dict['target'] = {
+                'cls': class_targets,
+                'bbox_offsets': bbox_offsets_targets,
+                'anchors': anchors
+            }
+
+        # Add network's raw output to prediction dict
+        prediction_dict['cls_pred'] = class_scores
+        prediction_dict['loc_pred'] = bbox_offsets
+
+        # We generate proposals when predicting, or when debug=True for
+        # generating visualizations during training.
+        if not is_training or self._debug:
+            proposals_creator = SSDProposal(
+                self._num_classes, self._config.model.proposals,
+                self._config.model.variances
+            )
+            proposals = proposals_creator(
+                class_probabilities, bbox_offsets, anchors,
+                tf.cast(tf.shape(image)[1:3], tf.float32)
+            )
+            prediction_dict['classification_prediction'] = proposals
+
+        # Add some non essential metrics for debugging
+        if self._debug:
+            prediction_dict['all_anchors'] = anchors
+            prediction_dict['cls_prob'] = class_probabilities
+            if gt_boxes:
+                # TODO does it make sense to return this when its an input
```

Comment:
It was used for image_vis and was called `gt_bboxes` not `gt_boxes`, so no, it doesn't make sense to keep it.